### PR TITLE
Adds the Prototype Bluespace RPED

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
[feature]
I'll get around the real deal device later. Lock that behind phazon Y/N?
This PR has been made by Discount Dan Gang.
Recolored the cool-looking lights to blue, I promise the real deal will have an attempt to be fancier (original content do not steal)
RPED
The device itself is more expensive than the normal one and costs diamonds but holds twice as much and ignores if machineries have the panels opened or not, because something something most bluespace stuff costs diamonds
:cl:
 * rscadd: Added the Prototype Bluespace Rapid Part Exchange Device to the Protolathe. It requires Engineering research of 5, Material research of 6, Data research of 3, and Bluespace research of 5. It can hold twice as much as a regular RPED and does not require a machinery's panel to be open to exchange parts. This one costs slightly more than a regular RPED but with diamond cost thrown into the mix.
* rscdel: The scientist screams are ONLY AVAILABLE FOR APRIL FOOLS. Sorry about that!

### CODE REVIEW

File: code/game/objects/items/weapons/storage/RPED.dm [hunk 1]
```diff
@@ -10,6 +10,16 @@
 	storage_slots = 50
 	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/misc_tools.dmi', "right_hand" = 'icons/mob/in-hand/right/misc_tools.dmi')
 	display_contents_with_number = TRUE
+	var/bluespace = 0
```
Comment (jknpj) [inline code/game/objects/items/weapons/storage/RPED.dm]: FALSE

File: code/game/objects/items/weapons/storage/RPED.dm [hunk 2]
```diff
@@ -10,6 +10,16 @@
...
+
+/obj/item/weapon/storage/bag/gadgets/part_replacer/discount_bluespace
+	name = "Prototype bluespace rapid part exchange device" //Alternative name: Discount BRPED, therefore denoted as DBRPED
+	desc = "Not as good as the real deal, but still good. This device is a better variant of the RPED that can hold twice as many parts and can function on machines that do not have their panels open."
+	icon_state = "DBRPED"
+	item_state = "DBRPED"
+	max_combined_w_class = 200
+	storage_slots = 100
+	bluespace = 1
```
Comment (jknpj) [inline code/game/objects/items/weapons/storage/RPED.dm]: TRUE

File: code/game/machinery/machinery.dm [hunk 3]
```diff
@@ -682,7 +682,7 @@ Class Procs:
 /obj/machinery/proc/exchange_parts(mob/user, obj/item/weapon/storage/bag/gadgets/part_replacer/W)
 	var/shouldplaysound = 0
 	if(component_parts)
-		if(panel_open)
+		if(panel_open || W.bluespace == 1)
```
Comment (jknpj) [inline code/game/machinery/machinery.dm]: W.bluespace would be enough, since it's a binary check.

**Comment by ghost** [comment]
I think this is a very nice middle-ground between the current RPED with its relatively low storage, and TG's Bluespace RPED, which just just feels like a symptom of their issue of everything having to have a bigger and better version _because it's cooooool!_

If you're really adamant on adding it at a later point, though, I think it should require something ridiculous like Bluespace 9\~11 and/or Materials 8~10, mostly because yeah, that requires Phazon and means they won't get it without at least some minor display of competency. 

That being said, I'm just heavily biased against TG's many _"everything needs a bigger and better version!"_ additions, so take it with a grain of salt.

**Comment by Lunstra** [comment]
This lets me hold an entire medbay worth of parts in a single RPED for autism projects in maint, this is good.

**Comment by ghost** [comment]
Did you test this with the autolathe (not protolathe)? It recycles things when hit by them.

**Comment by Kurfursten** [comment]
Why does it mess up for one frame aaaa

**Comment by MadmanMartian** [comment]
Up that bluespace tech cost, those are rookie numbers

**Comment by boy2mantwicethefam** [comment]
@Emilc4h I'll keep that in mind
@codeungabunga I checked with the regular RPED and the RPED just checks the machinery's components. Bluespace RPED works in exchanging the Autolathe's components. I legitimately don't think there's a function to throw all the stuff you have into the Autolathe using an RPED
@Kurfursten I moused over the sprite in DMI and recorded the part as a gif :joy: 
@MadmanMartian It's a prototype/discount version, the real deal RPED will have a much higher requirement

**Comment by boy2mantwicethefam** [comment]
Do tell me if you notice anything wrong with the storage.dm

**Pull request merged**